src/main/cpp/src/ftos_converter.cu

```diff
@@ -0,0 +1,1569 @@
+// Copyright 2018 Ulf Adams
+//
+// The contents of this file may be used under the terms of the Apache License,
+// Version 2.0.
+//
+//    (See accompanying file LICENSE-Apache or copy at
+//     http://www.apache.org/licenses/LICENSE-2.0)
+//
+// Alternatively, the contents of this file may be used under the terms of
+// the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE-Boost or copy at
+//     https://www.boost.org/LICENSE_1_0.txt)
+//
+// Unless required by applicable law or agreed to in writing, this software
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuda/std/climits>
+#include <cuda/std/limits>
+#include <cuda/std/type_traits>
+#include <cuda/std/cassert>
+#include <cuda/std/cstdint>
+
+namespace spark_rapids_jni {
+
+namespace detail {
+namespace {
+
+// d2s.c from ryu
+// A floating decimal representing m * 10^e.
+typedef struct floating_decimal_64 {
+  uint64_t mantissa;
+  // Decimal exponent's range is -324 to 308
+  // inclusive, and can fit in a short if needed.
+  int32_t exponent;
+} floating_decimal_64;
+
+// f2s.c from ryu
+// A floating decimal representing m * 10^e.
+typedef struct floating_decimal_32 {
+  uint32_t mantissa;
+  // Decimal exponent's range is -45 to 38
+  // inclusive, and can fit in a short if needed.
+  int32_t exponent;
+} floating_decimal_32;
+
+struct ftos_converter {
+
+  //===== constants from ryu =====
+
+  // These tables are generated by PrintDoubleLookupTable.
+  static constexpr unsigned int DOUBLE_POW5_INV_BITCOUNT = 125;
+  static constexpr unsigned int DOUBLE_POW5_BITCOUNT = 125;
+  static constexpr unsigned int FLOAT_POW5_INV_BITCOUNT = (DOUBLE_POW5_INV_BITCOUNT - 64);
+  static constexpr unsigned int FLOAT_POW5_BITCOUNT = (DOUBLE_POW5_BITCOUNT - 64);
+  static constexpr unsigned int DOUBLE_MANTISSA_BITS = 52;
+  static constexpr unsigned int DOUBLE_EXPONENT_BITS = 11;
+  static constexpr unsigned int DOUBLE_BIAS = 1023;
+  static constexpr unsigned int FLOAT_MANTISSA_BITS = 23;
+  static constexpr unsigned int FLOAT_EXPONENT_BITS = 8;
+  static constexpr unsigned int FLOAT_BIAS = 127;
+
+  const uint64_t DOUBLE_POW5_INV_SPLIT2[15][2] = {
```

Comment:
Although the code was implemented somewhere else, we should enforce our style to maintain consistency. In our convention, we use `east const` style:
```suggestion
uint64_t const` ...
```

Reply:
Probably you should search for `const` in this file and change the `const west` style into `east const` style for consistency.